src/cli/mod.rs

```diff
@@ -92,7 +93,38 @@ pub struct CkbRoArgs {
         value_parser = value_parsers::NetworkTypeValueParser,
         default_value = "testnet"
     )]
-    pub network: NetworkType,
+    pub(crate) network: NetworkType,
+}
+
+#[derive(Args)]
+#[group(multiple = false)]
+pub struct FeeRateArgs {
+    /// The fixed fee rate for CKB transactions.
+    #[arg(
+        group = "fixed-fee-rate",
+        conflicts_with = "dynamic-fee-rate",
+        long = "ckb-fee-rate",
+        default_value = "1000"
+    )]
+    fixed_value: u64,
+
+    /// [Experimental] Enable dynamic fee rate for CKB transactions.
+    ///
+    /// The actual fee rate will be the `median` fee rate which is fetched through the CKB RPC method `get_fee_rate_statistics`.
+    ///
+    /// ### Warning
+    ///
+    /// Users have to make sure the remote CKB node they used are trustsed.
+    ///
+    /// For security, there is a hard limit of the dynamic ckb fee rate: `1_0000_0000`.
+    ///
+    /// Ref: <https://github.com/nervosnetwork/ckb/tree/v0.114.0/rpc#method-get_fee_rate_statistics>
+    #[arg(
+        group = "dynamic-fee-rate",
+        conflicts_with = "fixed-fee-rate",
+        long = "enable-dynamic-ckb-fee-rate"
+    )]
+    enable_dynamic: bool,
```

Comment:
LGTM.
It's flexible since a BTC-SPV-service maintainer can choose to enable it or not.